starfish/imagestack/imagestack.py

```diff
@@ -659,24 +659,26 @@ def _build_slice_list(
 
         return tuple(slice_list), axes
 
-    def _iter_indices(self, indices: Set[Indices]={Indices.ROUND, Indices.CH}
-                      ) -> Iterator[Mapping[Indices, int]]:
+    def _iter_indices(self, indices: Set[Indices]=None) -> Iterator[Mapping[Indices, int]]:
         """Iterate over provided indices
 
         Parameters
         ----------
         indices : Set[Indices]
-            The set of Indices to be iterated over
+            The set of Indices to be iterated over (default={Indices.ROUND, Indices.CH}).
 
         Yields
         ------
         Dict[str, int]
             Mapping of dimension name to index
 
         """
-        ranges = [np.arange(self.shape[ind]) for ind in indices]
+        if indices is None:
+            indices = {Indices.ROUND, Indices.CH}
```

Comment:
how is this different from have the default value be {Indices.ROUND, Indices.CH}?

Reply:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
oh whaaaa wow I'm learning so much about python today